nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/AbstractTestAnnotationNotification.java

```diff
@@ -153,23 +158,53 @@ protected void captureAndVerifyAnnotationEventNotification(Supplier<DocumentMode
             assertEquals(1, handledEvents.size());
             Event expectedEvent = handledEvents.get(0);
             assertEquals(annotationEventType, expectedEvent.getName());
-            checkDocumentEventContext(expectedEvent, annotationDocumentModel, annotatedDocumentModel);
-            checkReceivedMail(emailsResult.getMails(), annotationDocumentModel, annotatedDocumentModel,
-                    handledEvents.get(0), annotationEventType);
+
+            checkDocumentEventContext(expectedEvent, annotationDocumentModel, annotatedDocumentModel,
+                    annotatedDocumentModel);
+
+            // Deduct only the emails sent since the last time
+            List<SmtpMailServerFeature.MailMessage> previousMails = new ArrayList<>(mails);
+            mails.clear();
+            mails.addAll(emailsResult.getMails()
+                                     .stream()
+                                     .filter(Predicate.not(previousMails::contains))
+                                     .collect(Collectors.toList()));
+            checkReceivedMail(mails, annotationDocumentModel, annotatedDocumentModel, handledEvents.get(0),
+                    annotationEventType);
         }
     }
 
+    @Test
+    public void shouldNotifyWithTheRightAnnotatedDocument() {
+        // First comment
+        captureAndVerifyAnnotationEventNotification(() -> {
+            Annotation createdAnnotation = createAnnotationAndAddSubscription("CommentAdded", "Creation");
+            return session.getDocument(new IdRef(createdAnnotation.getId()));
+        }, COMMENT_ADDED, DOCUMENT_CREATED);
+
+        // Reply
+        captureAndVerifyAnnotationEventNotification(() -> {
+            Comment reply = createAnnotation(annotatedDocumentModel);
```

Comment:
AFAIK, this does not create a "reply", but just a new annotation on `annotatedDocumentModel`, am I wrong?

Reply:
Yes, wrong refactoring